subsys/mgmt/mcumgr/grp/img_mgmt/Kconfig

```
@@ -61,6 +61,39 @@ config MCUMGR_GRP_IMG_VERBOSE_ERR
 	  Add additional "rsn" key to SMP responses, where provided, explaining
 	  non-0 "rc" codes.
 
+config MCUMGR_GRP_IMG_ALLOW_CONFIRM_NON_ACTIVE_IMAGE_SECONDARY
+	bool "Allow to confirm secondary slot of non-active image"
+	default y
+	help
+	  Allows to confirm secondary (non-active) slot of non-active image.
+	  Normally it should not be allowed to confirm any slots of non-active
+	  image, via MCUmgr commands, to prevent confirming something that is
+	  broken and may not boot in other slot; instead application should
+	  have means to test and confirm the image.
+	  Unfortunately due to logic bug it has been allowed to confirm
+	  secondary slot of non-active image so this option is set to y
+	  by default.
+
+config MCUMGR_GRP_IMG_ALLOW_CONFIRM_NON_ACTIVE_IMAGE_ANY
+	bool "Allow to confirm slots of non-active image"
+	default n
+	select MCUMGR_GRP_IMG_ALLOW_CONFIRM_NON_ACTIVE_IMAGE_SECONDARY
+	help
+	  Allows to confirm any slot of non-active image.
+	  Normally it should not be allowed to confirm any slots of non-active
+	  image, via MCUmgr commands, to prevent confirming something that is
+	  broken and may not boot in other slot; instead application should
+	  have means to test and confirm the image.
+
+if !MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP
+config MCUMGR_GRP_IMG_ALLOW_ERASE_PENDING
+	bool "Allow to erase pending slot"
+	help
+	  Allows to erase secondary slot set for test or confirmed; this allows
+	  to erase slot that has already been set for next boot but there has been
+	  no reset yet so it has not yet been swapped.
```

```suggestion
	  Allows erasing secondary slot which is marked for test or confirmed; this allows
	  erasing slots that have been set for next boot but the device has not
	  reset yet, so has not yet been swapped.
```
Also this refers to secondary but in the case of direct XIP it could be primary, maybe "opposite" slot would be useful, I think all Kconfig options have this issue.